### ISSUE TITLE
Besu problems

### ISSUE THREAD

**Issue by holiman** [ethereum/hive#259]
Current problem: 
```

Using the following genesis
{
  "coinbase": "0x8888f1f195afa192cfee860698584c030f4c9db1",
  "difficulty": "0x020000",
  "extraData": "0x42",
  "gasLimit": "0x2fefd8",
  "mixHash": "0x2c85bcbce56429100b2108254bb56906257582aeafcbd682bc9af67a9f5aee46",
  "nonce": "0x78cc16f7b4f65485",
  "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "timestamp": "0x54c98c81",
  "alloc": {
    "a94f5374fce5edbc8e2a8697c15331677e6ebf0b": {
      "balance": "0x09184e72a000"
    }
  },
  "config": {
    "ethash": {},
    "chainID": 1,
    "homesteadBlock": 0,
    "daoForkBlock": 2000,
    "eip150Block": 0,
    "eip158Block": 0,
    "eip155Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "constantinopleFixBlock": 0
  }
}
Setting logging level to INFO
2020-04-14 16:34:56.657+00:00 | main | INFO  | Besu | Starting Besu version: besu/v1.4.4-dev-5867a4b2/linux-x86_64/oracle_openjdk-java-11
2020-04-14 16:34:59.699+00:00 | main | INFO  | StaticNodesParser | StaticNodes file /opt/besu/static-nodes.json does not exist, no static connections will$
2020-04-14 16:34:59.753+00:00 | main | INFO  | Besu | Connecting to 0 static nodes.
2020-04-14 16:35:12.086+00:00 | main | INFO  | KeyPairUtil | Generated new key 0x26e3008b7ef472175a6aef9925fc30d56ac5de1cfa02035b09422b4b9e15a72e96aca9867$
2020-04-14 16:35:12.580+00:00 | main | INFO  | RocksDBKeyValueStorageFactory | No existing database detected at /opt/besu. Using version 1
Genesis Config Error: 'TangerineWhistle' is scheduled for block 0 but it must be on or after block 2000.
Usage:

```
@shemnon any idea why it rejects the config?

**Comment by shemnon** [comment]
We consider Dao a proper fork and require that Tangerine Whistle follow it.  The `eip15xBlock` series.  The same goes for byzantium, contstantinople, and constantinopleFix.

**Comment by shemnon** [comment]
i.e. we expect this at the very least...

```
   "homesteadBlock": 0,
    "daoForkBlock": 2000,
    "eip150Block": 2000,
    "eip158Block": 2000,
    "eip155Block": 2000,
    "byzantiumBlock": 2000,
    "constantinopleBlock": 2000,
    "constantinopleFixBlock": 2000
```

**Comment by holiman** [comment]
Is it possible to leave the dao fork out? How does private chains do (no dao)?

**Comment by shemnon** [comment]
You can leave the dao fork out.  But if it exists the following forks need to be set after.

**Comment by holiman** [comment]
Yep, got it to work now, running in to other problems. Will dump it here, but I'm doing hive work on a separate machine which isn't logged in to github

**Comment by holiman** [comment]
So, here's the next problem
Importing some blocks

```
Using the following genesis
{
  "coinbase": "0x8888f1f195afa192cfee860698584c030f4c9db1",
  "difficulty": "0x020000",
  "extraData": "0x42",
  "gasLimit": "0x2fefd8",
  "mixHash": "0x2c85bcbce56429100b2108254bb56906257582aeafcbd682bc9af67a9f5aee46",
  "nonce": "0x78cc16f7b4f65485",
  "parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
  "timestamp": "0x54c98c81",
  "alloc": {
    "a94f5374fce5edbc8e2a8697c15331677e6ebf0b": {
      	"balance": "0x09184e72a000"
    }
  },
  "config": {
    "ethash": {},
    "chainID": 1,
    "homesteadBlock": 0,
    "eip150Block": 0,
    "eip158Block": 0,
    "eip155Block": 0,
    "byzantiumBlock": 0,
    "constantinopleBlock": 0,
    "constantinopleFixBlock": 0
  }
}
Loading remaining individual blocks...
```
We invoke `besu` with the given genesis, and ask it to import the next block: 
```
invoking /opt/besu/bin/besu  --genesis-file=/etc/besu/genesis.json blocks import --from=/blocks/0001.rlp
2020-04-15 17:43:33.937+00:00 | main | INFO  | BlocksSubCommand | Import RLP block data from /blocks/0001.rlp
2020-04-15 17:43:34.645+00:00 | main | INFO  | KeyPairUtil | Generated new key 0x290c071ad2ee67dada0c3e35e7e4686b5b67661de7a1253e891a5c73a9924d7c7a7141d3f$
2020-04-15 17:43:34.675+00:00 | main | INFO  | RocksDBKeyValueStorageFactory | No existing database detected at /opt/besu. Using version 1
2020-04-15 17:43:34.961+00:00 | main | INFO  | ProtocolScheduleBuilder | Protocol schedule created with milestones: [ConstantinopleFix: 0]
picocli.CommandLine$ExecutionException: Error while running command (org.hyperledger.besu.cli.subcommands.blocks.BlocksSubCommand$ImportSubCommand@61cd1c7$
        at picocli.CommandLine.executeUserObject(CommandLine.java:1778)
        at picocli.CommandLine.access$900(CommandLine.java:145)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2141)
        at picocli.CommandLine$RunLast.handle(CommandLine.java:2108)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1968)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:2322)
        at org.hyperledger.besu.cli.util.ConfigOptionSearchAndRunHandler.handle(ConfigOptionSearchAndRunHandler.java:61)
        at org.hyperledger.besu.cli.util.ConfigOptionSearchAndRunHandler.handle(ConfigOptionSearchAndRunHandler.java:31)
        at picocli.CommandLine$AbstractParseResultHandler.handleParseResult(CommandLine.java:1968)
        at picocli.CommandLine.parseWithHandlers(CommandLine.java:2322)
        at org.hyperledger.besu.cli.BesuCommand.parse(BesuCommand.java:1053)
        at org.hyperledger.besu.cli.BesuCommand.parse(BesuCommand.java:922)
        at org.hyperledger.besu.Besu.main(Besu.java:49)
Caused by: java.lang.IllegalStateException: Block 1 does not connect to the existing chain. Current chain head 0
        at org.hyperledger.besu.chainimport.RlpBlockImporter.lambda$lookupPreviousHeader$4(RlpBlockImporter.java:200)
        at java.base/java.util.Optional.orElseThrow(Optional.java:408)
        at org.hyperledger.besu.chainimport.RlpBlockImporter.lookupPreviousHeader(RlpBlockImporter.java:197)
        at org.hyperledger.besu.chainimport.RlpBlockImporter.importBlockchain(RlpBlockImporter.java:94)
        at org.hyperledger.besu.cli.subcommands.blocks.BlocksSubCommand$ImportSubCommand.importRlpBlocks(BlocksSubCommand.java:231)
        at org.hyperledger.besu.cli.subcommands.blocks.BlocksSubCommand$ImportSubCommand.run(BlocksSubCommand.java:164)
        at picocli.CommandLine.executeUserObject(CommandLine.java:1769)
        ... 12 more
```
Is it possible to see what besu considers the genesis hash to be?

**Comment by holiman** [comment]
And a different problem, when trying to get Besu to sync from a geth-instance: 
```
starting main client: /opt/besu/bin/besu  --network-id=1 --sync-mode=FAST --logging=INFO --genesis-file=/etc/besu/genesis.json --graphql-http-enabled --graphql-http-host=0.0.0.0 --host-whitelist=* --rpc-http-enabled --rpc-http-api=ETH,NET,WEB3,ADMIN --rpc-http-host=0.0.0.0
Setting logging level to INFO
2020-04-20 12:12:08.833+00:00 | main | INFO  | Besu | Starting Besu version: besu/v1.4.4-dev-74940d50/linux-x86_64/oracle_openjdk-java-11
2020-04-20 12:12:09.109+00:00 | main | INFO  | StaticNodesParser | StaticNodes file /opt/besu/static-nodes.json does not exist, no static connections will be created.
2020-04-20 12:12:09.109+00:00 | main | INFO  | Besu | Connecting to 0 static nodes.
2020-04-20 12:12:09.780+00:00 | main | INFO  | KeyPairUtil | Generated new key 0x62aab8ef2eae38a1348dcb3b050ace5b66a3e3a415907b2b2a4dc28cc525ecd138c532c227c2429bf8a37f249273ea65e38e6b89b3049e2bb00da5048cd0f4cb and stored it to /opt/besu/key
2020-04-20 12:12:09.799+00:00 | main | INFO  | RocksDBKeyValueStorageFactory | No existing database detected at /opt/besu. Using version 1
2020-04-20 12:12:10.282+00:00 | main | INFO  | ProtocolScheduleBuilder | Protocol schedule created with milestones: [Byzantium: 0]
2020-04-20 12:12:10.410+00:00 | main | INFO  | RunnerBuilder | Detecting NAT service.
2020-04-20 12:12:10.947+00:00 | main | INFO  | Runner | Starting Ethereum main loop ... 
2020-04-20 12:12:10.947+00:00 | main | INFO  | DockerNatManager | Starting docker NAT manager.
2020-04-20 12:12:10.947+00:00 | main | INFO  | NetworkRunner | Starting Network.
2020-04-20 12:12:11.060+00:00 | nioEventLoopGroup-2-1 | INFO  | RlpxAgent | P2P RLPx agent started and listening on /0.0.0.0:30303.
2020-04-20 12:12:11.061+00:00 | main | INFO  | PeerDiscoveryAgent | Starting peer discovery agent on host=0.0.0.0, port=30303
2020-04-20 12:12:11.117+00:00 | vert.x-eventloop-thread-1 | INFO  | VertxPeerDiscoveryAgent | Started peer discovery agent successfully, on effective host=0.0.0.0 and port=30303
2020-04-20 12:12:11.119+00:00 | vert.x-eventloop-thread-1 | INFO  | PeerDiscoveryAgent | P2P peer discovery agent started and listening on /0.0.0.0:30303
2020-04-20 12:12:11.152+00:00 | main | INFO  | DefaultP2PNetwork | Enode URL enode://62aab8ef2eae38a1348dcb3b050ace5b66a3e3a415907b2b2a4dc28cc525ecd138c532c227c2429bf8a37f249273ea65e38e6b89b3049e2bb00da5048cd0f4cb@127.0.0.1:30303
2020-04-20 12:12:11.154+00:00 | main | INFO  | DefaultSynchronizer | Starting synchronizer.
2020-04-20 12:12:11.155+00:00 | main | INFO  | FastSyncDownloader | Starting fast sync.
2020-04-20 12:12:11.157+00:00 | main | INFO  | WaitForPeersTask | Waiting for 5 peers to connect. 0 peers currently connected.
2020-04-20 12:12:11.163+00:00 | main | INFO  | JsonRpcHttpService | Starting JsonRPC service on 0.0.0.0:8545
2020-04-20 12:12:11.209+00:00 | vert.x-eventloop-thread-5 | INFO  | JsonRpcHttpService | JsonRPC service started and listening on 0.0.0.0:8545
2020-04-20 12:12:11.209+00:00 | main | INFO  | GraphQLHttpService | Starting GraphQL HTTP service on 0.0.0.0:8547
2020-04-20 12:12:11.214+00:00 | vert.x-eventloop-thread-7 | INFO  | GraphQLHttpService | GraphQL HTTP service started and listening on 0.0.0.0:8547
2020-04-20 12:12:11.215+00:00 | main | INFO  | Runner | Ethereum main loop is up.
2020-04-20 12:12:11.224+00:00 | main | INFO  | AutoTransactionLogBloomCachingService | Starting auto transaction log bloom caching service.
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.fasterxml.jackson.databind.util.ClassUtil (file:/opt/besu/lib/jackson-databind-2.10.1.jar) to constructor java.util.OptionalLong()
WARNING: Please consider reporting this to the maintainers of com.fasterxml.jackson.databind.util.ClassUtil
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by holiman** [comment]
The problem with genesis hash is resolved by #266
